rd-kt/rd-core/src/commonMain/kotlin/com/jetbrains/rd/util/reactive/Signal.kt

```diff
@@ -46,10 +48,18 @@ class Signal<T> : ISignal<T> {
                         if (arr.contains(handler)) throw IllegalArgumentException("Duplicate handler: $handler")
                         arr.insert(handler, arr.size)
                     }
+                    if (queue.get().size == 10_000) {
+                        logger.warn { "Added over 10k handlers to a signal" }
+                    }
                 },
                 {
-                    queue.getAndUpdate { arr ->
-                        arr.remove (handler).apply { if (equals(arr)) throw IllegalArgumentException("No handler: $handler") }
+                    val elapsed = measureTimeMillis {
+                        queue.getAndUpdate { arr ->
+                            arr.remove(handler).apply { if (equals(arr)) throw IllegalArgumentException("No handler: $handler") }
+                        }
+                    }
+                    if (elapsed > 1000) {
+                        logger.warn { "Signal handler removal took longer than one second" }
```

Comment:
I'm sorry, I'm not super familiar with our architecture, but could we print some useful information about the signal? E.g. its identifier? Smth like `model.foo.bar[0]`?

Otherwise, I'm not sure this warning would be super useful for us. We're almost sure that some signal handler removals take longer that one second; confirming this fact wouldn't do much for us.

Reply:
Unfortunately no. This signal is a local-only primitive (as opposed to RdSignal that is usually bound to a protocol tree), so no information is available. I'll remove this one then.

Reply:
I suggest to leave it here. Just as an additional confirmation of our main hypothesis, it will work